pkg/adapter/v2/cloudevents.go

```diff
@@ -113,6 +116,24 @@ type client struct {
 
 var _ cloudevents.Client = (*client)(nil)
 
+// SendCloudEvent prepares logging while sending an event for APIServerSource and PingSource
+// and sends the event to client.Send
+func SendCloudEvent(ctx context.Context, event cloudevents.Event, logger *zap.SugaredLogger, ce cloudevents.Client) {
+	event = event.Clone()
```

Comment:
Cloning is not needed for APIServerSource but needed for PingSource. I would remove it here and keep it in the PingSource runner.